src/utils/url.js

```diff
@@ -26,6 +24,33 @@ function validateDestUrl (url) {
     }
 }
 
+export function parseResourceType (resourceType) {
+    if (!resourceType) {
+        return {
+            isIframe: false,
+            isForm:   false,
+            isScript: false
+        };
+    }
+
+    return {
+        isIframe: resourceType.indexOf('i') !== -1,
+        isForm:   resourceType.indexOf('f') !== -1,
+        isScript: resourceType.indexOf('s') !== -1
```

Comment:
This test lacks statistical significance and uses low-res timers. you jsperf.com for benchmarking: https://jsperf.com/regexp-indexof-perf